api/server/server.go

```diff
@@ -1137,8 +1137,17 @@ func (s *Server) getContainersByName(version version.Version, w http.ResponseWri
 		return fmt.Errorf("Missing parameter")
 	}
 
+	if err := parseForm(r); err != nil {
+		return err
+	}
+
+	no_err := r.Form.Get("no_err")
 	containerJSON, err := s.daemon.ContainerInspect(vars["name"])
 	if err != nil {
+		if strings.Contains(strings.ToLower(err.Error()), "no such id") && no_err == "1" {
+			w.WriteHeader(http.StatusNoContent)
```

Comment:
I don't think drop this 'error' down to 'info' or 'debug' make sense, it may be in this scenario, but what if uses use this cli in their applications just want to inspect container, they should got an error message if they sent a wrong container name, not some 'info' thing.

For #13187, last time I saw it was weeks ago, I remember people don't want add more options and I still see this daemon error today, so thought we can fix it in this way. I'm glad to see they finally have conclusion on this, I'm +1 on add `--type`, that fix the problem that container and image have the same name, but if we still want inspect one name for both image and container (seems necessary for backward compatibility), I think we still need this fix. Let's see where #13187 will go, if it can fix this problem, then we can close this one.

BTW, I do agree 404 is the right way to deal with 'not exist container', but client will take 404 as error and show error message, and I think hack this is also not decent, just made a choice.